pic-sure-auth-services/src/main/java/edu/harvard/hms/dbmi/avillach/auth/rest/UserService.java

```diff
@@ -442,4 +443,86 @@ private void checkAssociation(List<User> users) throws ProtocolException{
         }
     }
 
+    /**
+     * Create or update a user record, based on the FENCE user profile, which is in JSON format.
+     *
+     * @param node User profile, as it is received from Gen3 FENCE, in JSON format
+     * @return User The actual entity, as it is persisted (if no errors) in the PSAMA database
+     */
+    public User createUserFromFENCEProfile(JsonNode node) {
+        User actual_user = null;
+        try {
+            User user_to_look_up = new User().setSubject("fence|"+node.get("user_id"));
+
+            if (userRepo == null) {
```

Comment:
Throwing ApplicationException will be much better since it will be caught automatically by jax rs ApplicationExceptionMapper

Reply:
Fair enough. This is cleared out. Looks much better